logstash-core/lib/logstash/modules/kibana_config.rb

```diff
@@ -0,0 +1,104 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging"
+
+require_relative "file_reader"
+require_relative "kibana_resource"
+require_relative "kibana_base_resource"
+
+module LogStash module Modules class KibanaConfig
+  include LogStash::Util::Loggable
+
+  ALLOWED_DIRECTORIES = ["search", "visualization"]
+
+  attr_reader :index_name
+
+  # We name it `modul` here because `module` has meaning in Ruby.
+  def initialize(modul, settings)
+    @directory = ::File.join(modul.directory, "kibana")
+    @name = modul.module_name
+    @settings = settings
+    @index_name = settings.fetch("dashboards.kibana_index", ".kibana")
+  end
+
+  def dashboards
+    # there can be more than one dashboard to load
+    filenames = FileReader.read_json(dynamic("dashboard"))
+    filenames.map do |filename|
+      KibanaResource.new(@index_name, "dashboard", dynamic("dashboard", filename))
+    end
+  end
+
+  def index_pattern
+    pattern_name = "#{@name}-*"
+    default_index_json = '{"defaultIndex": "#{pattern_name}"}'
+    default_index_content_id = @settings.fetch("index_pattern.kibana_version", "5.4.0") # make this 5.5.0
```

Comment:
@guyboertje Isn't this about how we manage semver? We should just care about major?